tools/wake-format/emitter.cpp

```diff
@@ -78,6 +78,27 @@ static inline bool is_expression(cst_id_t type) {
          type == CST_INTERPOLATE || type == CST_MATCH || type == CST_REQUIRE || type == CST_PRIM;
 }
 
+static inline bool is_primary_term(wcl::doc_builder& builder, ctx_t ctx, CSTElement& node,
+                                   const token_traits_map_t& traits) {
```

Comment:
How does this relate to the simple-expression determination we were using for... `require` was it?  I know there was something which we were at least talking about allowing single lines for in some cases.  If that's a similar domain, it feels like it might be easier for people to reason about what format to use themselves if we use a consistent trigger for the exceptions -- is there anything critical there which we'd lose by switching to use `is_primary_term` across the board?

Reply:
These are different. This is a syntactic category from the grammar. The "simple" category is more or less "single token" and has no exact analog in the grammar.

You can think of this category as the category of expressions which are clearly on inspection higher precedence than any prefix operator.